core/src/main/scala/kafka/tools/StreamsResetter.java

```diff
@@ -277,6 +284,28 @@ private int maybeResetInputAndSeekToEndIntermediateTopicOffsets(final Map consum
             }
         }
 
+        if (!notFoundInputTopics.isEmpty()) {
+            System.out.println("Following input topics are not found, skipping them");
+            for (final String topic : notFoundInputTopics) {
+                System.out.println("Topic: " + topic);
+            }
+            topicNotFound = EXIT_CODE_ERROR;
+        }
+
+        if (!notFoundIntermediateTopics.isEmpty()) {
+            System.out.println("Following intermediate topics are not found, skipping them");
+            for (final String topic : notFoundIntermediateTopics) {
+                System.out.println("Topic:" + topic);
+            }
+            topicNotFound = EXIT_CODE_ERROR;
+        }
+
+        // Return early if there are no topics to reset (the consumer will raise an error if we
+        // try to poll with an empty subscription)
+        if (topicsToSubscribe.isEmpty()) {
```

Comment:
The early return is needed to prevent the consumer from raising an illegal state error in poll if there are no subscribed topics.

Reply:
Sounds right to me.

Reply:
Ack.